modules/representation_management/app/models/representation_management/accredited_entity_query.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module RepresentationManagement
+  class AccreditedEntityQuery
+    include ActiveModel::Model
+
+    def initialize(query_string)
+      @query_string = query_string
+    end
+
+    def results
+      (individuals + organizations).sort_by do |record|
+        levenshtein_distance(@query_string, record)
+      end.take(10)
+    end
+
+    private
+
+    def individuals
+      AccreditedIndividual.where('word_similarity(?, full_name) >= ?', @query_string, threshold)
+    end
+
+    def organizations
+      AccreditedOrganization.where('word_similarity(?, name) >= ?', @query_string, threshold)
+    end
+
+    def threshold
+      0.7
+    end
+
+    def levenshtein_distance(query, record)
+      text = record.is_a?(AccreditedIndividual) ? record.full_name : record.name
+      StringHelpers.levenshtein_distance(query, text)
```

Comment:
If we're ever open to adding more postgres extensions, hopefully we get `fuzzystrmatch` and then we could move this to the db

Reply:
You could add the extension if you'd like. Just create a migration:

```ruby
class AddFuzzystrmatchExtension < ActiveRecord::Migration[6.0]
  def change
    enable_extension 'fuzzystrmatch'
  end
end
```

Strong migrations might want you to do something here.

Reply:
I'd rather do it in PG than Ruby memory. If not we should probably limit results to a sane max (100? 1000?).

Reply:
As of right now we're querying both models for the closest matches based on Levenshtein distance with a limit of 10 results.  Then we're taking those 20 rep/org records and and sorting them again by Levenshtein distance but that sorting is in Ruby.

I'm comfortable sorting that in Ruby because we'll only have 20 items to sort.  I guess the alternative is to construct some kind of SQL query that can query both individuals and organizations then sort all of those results by Levenshtein distance.  Is that what you had in mind @jvcAdHoc and @ericboehs ?

Reply:
Update: All querying and sorting now happens in SQL.